tests/recovery/blobmanipulator/blobmanipulator.go

```diff
@@ -0,0 +1,269 @@
+
+	// assumption: the repo under test is in filesystem
+	err := bm.KopiaCommandRunner.ConnectRepo("filesystem", "--path="+bm.DataRepoPath)
+	if err != nil {
+		return "", err
+	}
+
+	blobIDList, _, _ := bm.KopiaCommandRunner.Run("blob", "list", "--json")
+	if blobIDList == "" {
+		return "", robustness.ErrNoOp
+	}
+
+	err = json.Unmarshal([]byte(blobIDList), &b)
+	if err != nil {
+		return "", err
+	}
+
+	blobToBeDeleted := ""
+	// Select the first pack blob in the list
+	for _, s := range b {
+		temp := string(s.BlobID)
+		if strings.HasPrefix(temp, "p") {
+			blobToBeDeleted = string(s.BlobID)
+			break
+		}
+	}
+
+	return blobToBeDeleted, nil
+}
+
+func (bm *BlobManipulator) getFileWriter() bool {
+	fw, err := fiofilewriter.New()
+	if err != nil {
+		if errors.Is(err, fio.ErrEnvNotSet) {
+			log.Println("Skipping recovery tests because FIO environment is not set")
+		} else {
+			log.Println("Error creating fio FileWriter:", err)
+		}
+
+		return false
+	}
+
+	bm.fileWriter = fw
+
+	return true
+}
+
+// RestoreGivenOrRandomSnapshot restores a given or a random snapshot from kopia repository into the provided target directory.
+func (bm *BlobManipulator) RestoreGivenOrRandomSnapshot(snapID, restoreDir string) (string, error) {
+	err := bm.KopiaCommandRunner.ConnectRepo("filesystem", "--path="+bm.DataRepoPath)
+	if err != nil {
+		return "", err
+	}
+
+	if snapID == "" {
+		// list available snaphsots
+		stdout, _, snapshotListErr := bm.KopiaCommandRunner.Run("snapshot", "list", "--json")
+		if snapshotListErr != nil {
+			return stdout, snapshotListErr
+		}
+
+		var s []snapshot.Manifest
+
+		err = json.Unmarshal([]byte(stdout), &s)
+		if err != nil {
+			return "", err
+		}
+
+		// Select a random snapshot to restore
+		snapID = string(s[rand.Intn(len(s))].ID)
+	}
+
+	_, msg, err := bm.KopiaCommandRunner.Run("snapshot", "restore", snapID, restoreDir)
+	if err != nil {
+		return msg, err
+	}
+
+	return "", nil
+}
+
+// SetUpSystemUnderTest connects or creates a kopia repo, writes random data in source directory,
+// creates snapshots of the source directory.
+func (bm *BlobManipulator) SetUpSystemUnderTest() {
+	err := bm.ConnectOrCreateRepo(bm.DataRepoPath)
+	if err != nil {
+		return
+	}
+
+	// create random data
+	bm.getFileWriter()
+
+	fileSize := 100
+	numFiles := 100
+
+	fileWriteOpts := map[string]string{
+		fiofilewriter.MaxDirDepthField:         strconv.Itoa(1),
+		fiofilewriter.MaxFileSizeField:         strconv.Itoa(fileSize),
+		fiofilewriter.MinFileSizeField:         strconv.Itoa(fileSize),
+		fiofilewriter.MaxNumFilesPerWriteField: strconv.Itoa(numFiles),
+		fiofilewriter.MinNumFilesPerWriteField: strconv.Itoa(numFiles),
+	}
+
+	ctx := context.Background()
+
+	_, err = bm.fileWriter.WriteRandomFiles(ctx, fileWriteOpts)
+	if err != nil {
+		return
+	}
+
+	// create snapshot of the data
+	snapPath := bm.fileWriter.DataDirectory(ctx)
+	log.Printf("Creating snapshot of directory %s", snapPath)
+
+	_, err = bm.TakeSnapshot(snapPath)
```

Comment:
Verify that snapshot succeeded:  either inspect or restore snapshot and verify that contents match results of `WriteRandomFiles`

Reply:
I will do this in a subsequent PR.